jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js

```diff
@@ -215,36 +244,41 @@ YUI.add('deployment-add-credentials', function() {
                   size="12" />
               </a>
             </div>
-            <form className="twelve-col-col">
+            <form className="twelve-col">
               <div className="six-col">
-                <label className="deployment-panel__label"
-                  htmlFor="credential-name">
-                  {credentialName}
-                </label>
-                <input className="deployment-panel__input"
-                  id="credential-name"
-                  placeholder="AWS_1"
-                  required="required"
-                  type="text"
-                  ref="templateName" />
-                <label className="deployment-panel__label"
-                  htmlFor="specify-region">
-                  Region
-                </label>
-                <input className="deployment-panel__input"
-                  id="specify-region"
+                <juju.components.DeploymentInput
+                  label={credentialName}
+                  placeholder="AWS-1"
+                  required={true}
+                  ref="templateName"
+                  validate={[{
+                    regex: /\S+/,
+                    error: 'This field is required.'
+                  }, {
+                    regex: /^[a-zA-Z0-9]([a-zA-Z0-9.-]*[a-zA-Z0-9])?$/,
+                    error: 'This field must only contain upper and lowercase' +
```

Comment:
Needs a space after "lowercase" or before "letters". Only noticed this when I was QAing.